### ISSUE TITLE
Add Ollama block

### ISSUE THREAD

**Issue by Lanhild** [baptisteArno/typebot.io#1406]
APIs like Ollama are fully compatible with the OpenAI spec, thus we should have the capabability of adding a custom endpoint in  OpenAI block credentials
![image](https://github.com/baptisteArno/typebot.io/assets/73610779/023c9c1c-8359-487f-8fcb-d5cb0c70b0e4)

**Comment by baptisteArno** [comment]
Ollama is local. I think that does not really make sense for a tool like Typebot which is 100% online?

What would be the use case?

**Comment by ZimaBlueee** [comment]
@baptisteArno Hi

Local large models are really important and are also a trend.

Please also add xinference, which supports more models than ollama and has more comprehensive functions.

Official documentation: https://inference.readthedocs.io/en/latest/index.html

**Comment by baptisteArno** [comment]
I don't think it is a good idea to offer the Ollama block since most people would like to consume it locally. Typebot by default is 100% online so it won't work with local ollama instances. Closing it for now.

**Comment by easaw** [comment]
Hi has this been deployed or is it still not planned?

A use case is that an organisation can use local AI to allow (sales) staff to ask queries about services and staff documentation which is complex, and is updated frequently. But the data must be internal only.

I can't find another way to do this without using extensive cloud resources (and yes, I am aware of Botpress, langflow, localAI, lobechat, anythingLLM, LMStudio, ActivePieces/ n8n, AsktheDoc, Docuchat, ChatDoc, Documind and PrivateGPT etc...)

**Comment by easaw** [comment]
@Lanhild I did think of one way to do this, not the best, but;

Change the hosts file on the machine running Typebot so that api.openai.com (or whatever) uses the local IP for your Ollama instead.

**Comment by rizwan95** [comment]
typebot must support ollama. It is the need of the hour.

**Comment by leadvic** [comment]
I'd love to deploy my own instance of Ollama in my servers with a reverse proxy and all of that and use it in combination with TypeBot, it's really sad it is not planned.

**Comment by baptisteArno** [comment]
Reopening as it make sense for new in-app AI features

**Comment by nise** [comment]
Are there any updates on this? 

The integration of Ollama API is a core requirement for processing personal data inline with the EU GDPR. The current approach with OpenAI requires tones of compliance documents to consider all legal aspects.

**Comment by Zelray** [comment]
I have a server, that's not even very expensive, and I can run a 12B parameter model that is perfect for my customer service, and a 3B parameter model for my RAG and they both fly! Mr. Baptiste, I'm really surprised you don't know about this. If someone is self-hosting typebot, they are also going to want to run Ollama- the two go hand-in-hand. I'll be happy to contribute!

**Comment by baptisteArno** [comment]
Typebot is now in maintenance mode, so I’m no longer planning new features, product improvements or large internal projects. I’m closing this to keep the issue tracker focused on bugs, security and compatibility. Thanks for the suggestion!
